### ISSUE TITLE
about inputs of SemiRankLoss

### ISSUE THREAD

**Issue by iTomxy** [PKU-ICST-MIPL/SSDH_TCSVT2017#4]
two questions:
1. what's the difference between `label` and `gndtruth`? is that `label` only contain the label for labeled data, while `gndtruth` contains both labeled & unlabeled part?
2. what's the `sim` ? seems that it comes from the file *flickr25k_semitrain_resize.txt*, but I don't understant the meaning of it, which range from `0` to `14`.

**Comment by PKU-ICST-MIPL** [comment]
the gndtruth and sim are only used for debug. 
the label contains the label for labeled, for unlabeled data, the label is set to zeros.

**Comment by iTomxy** [comment]
there are `39,000` lines in *flickr25k_semitrain_resize.txt* and `20,000` lines in *flickr25k_qteset_resize.txt*, but according to the paper, there are only `1,000` query samples and `24,000` retrieval samples.
how are the contents of these two lists organized?

**Comment by PKU-ICST-MIPL** [comment]
We choose 5000 labeled data and use the remaining 20,000 data as the test set. 
39000 contains 5000 labeled data and some unlabeled data from the test set. 
Besides, we repeat the 5000 labeled to balance the size of the labeled data in the training set.

**Comment by iTomxy** [comment]
Are those samples denoted by `0` in *flickr25k_semitrain_resize.txt* the unlabeled samples? I checked the amount of them and it's exactly `19,000`, which matches `25,000 - 1,000 - 5,000` (*#total - #query - #labeled*).

**Comment by LiuChaoXD** [comment]
have you successfully construct the h5 file?

**Comment by 0sunshine0** [comment]
What are the two margin parameters mp and mt in the loss of SSDH?
